apps/bpytop.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+# T&M Hansson IT AB © - 2020, https://www.hanssonit.se/
+
+# shellcheck disable=2034,2059
+true
+SCRIPT_NAME="BPYTOP"
+SCRIPT_EXPLAINER="BPYTOP is an amazing alternative to ressource-monitor software like top or htop."
```

Comment:
I mean this, SCRIPT_EXPLAINER is already there.

Reply:
Yes, but I changed the implementation recently so that an additional `msg_box "$SCRIPT_EXPLAINER"` is needed at the beginning of any script that has one.

Reply:
That seems unnecessary, wouldn't it be possible to only provide the variable at the top?

Reply:
Why did you change it? :thinking:

Reply:
Because it was bound to the explainer_popup which I removed. If you want to reintroduce it, we could bind it to install_popup and reinstall_remove_menu instead.

Reply:
> If you want to reintroduce it, we could bind it to install_popup and reinstall_remove_menu instead.

But there are some problems with that. Because of that I just did it this way.

Reply:
We could also reintroduce the explainer_popup function which could only show a msg_box with description if SCRIPT_EXPLAINER is set.
But I'd rather add a msg_box "$SCRIPT_EXPLAINER"

Reply:
Here is an example from the Remotedesktop script:
https://github.com/nextcloud/vm/blob/f0f21aabcd4ee7049febdd82ddcc074c2245b329/not-supported/remotedesktop.sh#L23-L24

Reply:
Well, it seems like double code.

First you set `SCRIPT_EXPLAINER` then you execute `SCRIPT_EXPLAINER` with `msg_box` - instead of just using the content in `SCRIPT_EXPLAINER``like plain text in the `msg_box` like:

Correct:
msg_box "This is the text of the `SCRIPT_EXPLAINER`"

Seems unessesary:
SCRIPT_EXPLAINER="This is the text of the `SCRIPT_EXPLAINER`"
msg_box "`$SCRIPT_EXPLAINER`"

Reply:
I don't think so because it makes the function of the script directly visible to the user that reads the script because it is on the top and labeled with script_explainer.
This wouldn't be as clear if we'd just use msg_box "text" anywhere down there.